File: insertion_sort.py
#insertion sort
def insertion_sort(array,size):
    for i in range(1,size):
        key=array[i]
        j=i-1
        while(j>-1 and key<array[j]):
            array[j+1]=array[j]
            j-=1
        array[j+1]=key
array=[10,3,2,5,4,7,6,8,9,1]
insertion_sort(array,10)
print(array)

File: sorting/insertion_sort.cpp
/*
    Insertion sort
    start with second position and put into right place on the left
    eg sorting cards
*/
#include<iostream>
using namespace std;
void insertion_sort(int *array,int size)
{
    int j=0,key;
    for(int i=1;i<size;i++)
    {
        key=array[i];
        j=i-1;
        while(j>-1 && array[j]>key)
        {
            array[j+1]=array[j];
            j-=1;
        }
        array[j+1]=key;
    }
}
int main()
{
    int array[10]={1,2,3,5,6,4,9,8,7,10};
    insertion_sort(array,10);
    for(int i=0;i<10;i++)
    {
        cout<<array[i]<<" ";
    }
    cout<<endl;
}
